### PR TITLE
feat(dart): generic translation

### PR DESCRIPTION
**Pull request by brandonspark**
## What:
This PR implements support for the Dart programming language.

## Why:
We currently only have raw trees, meaning we get _very_ limited matching capabilities, but nowhere near the standard level of support Semgrep can have for a language.

## How:
Implemented the translation.

There are a few TODOs, but I judged them to be things which are unlikely to crop up / relatively unimportant.

## Test plan:
`make test`

```
lang/dart/stats.json:3
   "global": {
     "name": "*",
     "parsing_rate": 0.9660407390632094,
     "line_count": 7535323,
     "error_line_count": 255894,
     "file_count": 42736,
     "error_file_count": 3885,
     "total_node_count": 231327789,
     "untranslated_node_count": 196495542
   },
```

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date (https://github.com/returntocorp/semgrep-docs/pull/1186)
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by aryx — CHANGES_REQUESTED** [review]
Please do not add semgrep-dart as-is; you should add it as a submodule instead

File: languages/dart/generic/Parse_dart_tree_sitter.ml [hunk 1]
```diff
@@ -0,0 +1,3870 @@
+(* Yoann Padioleau
```
Comment (aryx) [inline languages/dart/generic/Parse_dart_tree_sitter.ml]: put your name here brandon

**Review by aryx — CHANGES_REQUESTED** [review]
nope, still not moved properly.
Please try to review your own PR to catch those kind of issues.

**Comment by brandonspark** [comment]
@aryx ok, NOW it should be good i think

**Review by aryx — APPROVED** [review]
3800 LOC ...
I think you forgot to modify Unit_engine.ml and add the maturity test for Dart.

**Pull request merged**